core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java

```diff
@@ -171,7 +173,10 @@ public Set<Long> getReplicationLimitedFileIds() {
    * @return the root of the inode tree
    */
   public InodeDirectory getRoot() {
-    return mInodeStore.get(0).map(Inode::asDirectory).orElse(null);
+    if (mRoot == null) {
+      mRoot = mInodeStore.get(0).map(Inode::asDirectory).orElse(null);
```

Comment:
```suggestion
      synchronized (this) {
          if (mRoot == null) {
              mRoot = mInodeStore.get(0).map(Inode::asDirectory).orElse(null);
          }
      }
```
Also `mRoot` needs to be volatile.

Reply:
@dbw9580 `volatile` is expensive while read, as this place is a hotspot, so we don't want to involve a lock or `volatile` here if the currency access to `L178-L179` won't cause any issue.

So, we can try to find a scenario will encounter a issue by currency access.